app/src/vuex/modules/session.js

```diff
@@ -172,21 +206,37 @@ export default ({ apollo }) => {
       const allSessionAddresses = await dispatch("getAllSessionAddresses")
       commit("setAllSessionAddresses", allSessionAddresses)
 
-      state.externals.track(`event`, `session`, `sign-in`, sessionType)
+      state.externals.track(
+        `event`,
+        `session`,
+        `sign-in`,
+        sessionType,
+        accountType
+      )
     },
-    async signOut({ state, commit, dispatch }, networkId) {
+    async signOut({ state, commit, dispatch }, { address, networkId }) {
       state.externals.track(`event`, `session`, `sign-out`)
 
-      dispatch(`resetSessionData`, networkId)
+      dispatch(`resetSessionData`, { address, networkId })
       commit(`setSignIn`, false)
 
       // update session addresses
       const allSessionAddresses = await dispatch("getAllSessionAddresses")
       commit("setAllSessionAddresses", allSessionAddresses)
     },
-    resetSessionData({ commit, state }, networkId) {
+    resetSessionData({ commit, state }, { address, networkId }) {
+      const currentAddress = address || state.address
       state.history = ["/"]
       commit(`setUserAddress`, null)
+      if (currentAddress) {
+        // session_<networkId> (sessionKey) will store the current active session while session_<networkId>_<address> (historySessionKey)
+        // will serve as historical log in localStorage for all the ever initiated sessions
```

Comment:
should we remove the historic addresses? who is using them these days?

Reply:
the `cosmos-wallets` you mean? They contain the encrypted wallet. I wouldn't remove them: last possible resort to regain control over an account if they lose the seeds and it is an old account

Reply:
the session just associates a network and an address doesn't it? the keys are stored separate

Reply:
yes, that is how it works